go/vt/vtgate/planbuilder/physical/route_planning.go

```diff
@@ -919,12 +920,71 @@ func canMergeOnFilter(ctx *plancontext.PlanningContext, a, b *Route, predicate s
 	return rVindex == lVindex
 }
 
+func canMergeSubqueryOnColumnSelection(ctx *plancontext.PlanningContext, a, b *Route, predicate *sqlparser.ExtractedSubquery) bool {
+	left := predicate.OtherSide
+	opCode := predicate.OpCode
+	if opCode != int(engine.PulloutValue) && opCode != int(engine.PulloutIn) {
+		return false
+	}
+
+	lVindex := findColumnVindex(ctx, a, left)
+	if lVindex == nil || !lVindex.IsUnique() {
+		return false
+	}
+
+	rightSelection := extractSingleColumnSubquerySelection(predicate.Subquery)
+	if rightSelection == nil {
+		return false
+	}
+
+	rVindex := findColumnVindex(ctx, b, rightSelection)
+	if rVindex == nil {
+		return false
+	}
+	return rVindex == lVindex
```

Comment:
Do we need to find the column vindex given that `extractSingleColumnSubquerySelection` will return a column and we have a lVindex. Isn't there a way to check that the column is covered by this vindex. What would happen if we have two or more vindexes on the same column?

Reply:
I don't understand why someone would create two different unique vindexes on the same column. if that would happen, and the vschema would give us different vindexes when querying the two parts of the query, the worst thing that would happen would be that we don't merge the subquery into the outer part of the query, and that is not that bad

Reply:
Okay, makes sense.